packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -61,9 +61,9 @@ export function createInsightsMiddleware<
     $$internal = false,
   } = props;
 
-  let insightsClient: InsightsClient = _insightsClient || noop;
+  let insightsClient: InsightsClient & { needsToLoadInsightsClient?: boolean } =
```

Comment:
Shorter and less redundant, wdyt?

```suggestion
  let insightsClient: InsightsClient & { shouldLoad?: boolean } =
```

Reply:
I think that's a little unclear, as load could mean many things just like that, but committed `shouldAddScript` now, makes sense